dashboard/app/views/levels/_dialogHelper.js

```diff
@@ -35,14 +35,24 @@
     // Use our prefabricated dialog content.
     $(".modal-body").append($("#" + type + "-dialogcontent").clone(true));
 
-    // Clicking the okay button in the dialog box dismisses it.
+    // Clicking the okay button in the dialog box dismisses it, and calls the callback.
     $(".modal-body #ok-button").click(function () {
       dialog.hide();
+      callback && callback();
+    });
+
+    // Clicking the cancel button in the dialog box dismisses it.
+    $(".modal-body #cancel-button").click(function () {
+      dialog.hide();
     });
 
     dialog.show();
   }
 
+  window.showStartOverDialog = function(callback) {
```

Comment:
@joshlory you do raise a good point, which is that we should have a plan for getting rid of these global references. Last time we discussed this, we were torn over whether to pass callbacks in appOptions, which is good because it eliminates globals, but bad because another goal is for appOptions to be defined entirely in Ruby. I think passing callbacks via appOptions is a step in the right direction. I'll make a follow-up PR.